lib/agenda/find-and-lock-next-job.js

```diff
@@ -36,23 +36,19 @@ module.exports = function(jobName, definition, cb) {
   } else {
     // /**
     // * Query used to find job to run
-    // * @type {{$or: [*]}}
+    // * @type {{$and: [*]}}
     // */
     const JOB_PROCESS_WHERE_QUERY = {
-      $or: [{
+      $and: [{
         name: jobName,
-        lockedAt: null,
-        nextRunAt: {$lte: this._nextScanAt},
         disabled: {$ne: true}
       }, {
-        name: jobName,
-        lockedAt: {$exists: false},
```

Comment:
Do you know if this `lockedAt: {$exists: false}` should be one of the options in `$or`?

Reply:
AFAIK (from the top of my head) your can check both existence and null equality with "$eq: null". Whereas the "$exists: false" checks the existence only.

But I doesn't trust my memory too much. 🙂 So, please double check.

In other words, the PR is okay IMO.

Reply:
Sounds good! Thanks.

Reply:
This has caused a migration bug from v2->v4.
Here are the details: https://github.com/agenda/agenda/issues/1238